slam_toolbox/src/merge_maps_kinematic.cpp

```diff
@@ -234,9 +234,12 @@ bool MergeMapsKinematic::mergeMapCallback(
 /*****************************************************************************/
 {
   ROS_INFO("Merging maps!");
+  std::unique_ptr<karto::Mapper> merged_mapper = std::make_unique<karto::Mapper>();
+  std::unique_ptr<karto::Dataset> merged_dataset = std::make_unique<karto::Dataset>();
 
   // transform all the scans into the new global map coordinates 
   int id = 0;
+  bool processed = false;
```

Comment:
You don't use this variable. You can just ignore the return type if you're not checking its validity.

Reply:
Hello, done. 

And I am developing the lifelong merge, I am planning to add a new service something like  "lifelong merge", it works as lifelong to decay old node when merge pose graph , if you want, I can create another PR after I finish.